src/Core/Commander/Providers/WMS_Provider.js

```diff
@@ -30,71 +32,221 @@ define('Core/Commander/Providers/WMS_Provider', [
+
+                if(texture)
+                    result.texture = texture;
+                else
+                {
+                    result.texture = new THREE.Texture(image);
+                    result.texture.needsUpdate = true;
+                    result.texture.generateMipmaps = false;
+                    result.texture.magFilter = THREE.LinearFilter;
+                    result.texture.minFilter = THREE.LinearFilter;
+                    result.texture.anisotropy = 16;
+                    result.texture.url = url;
+                   // result.texture.layerId = layerId;
+
+                    this.cache.addRessource(url, result.texture);
+                }
+
+                return result;
+
+            }.bind(this)).catch(function(/*reason*/) {
+                    result.texture = null;
+
+                    return result;
+                });
+
+        };
+
+        WMS_Provider.prototype.getElevationTexture = function(bbox, layer) {
+            var url = this.url(bbox,layer);
+
+            // TODO: this is not optimal: if called again before the IoDriver resolves, it'll load the XBIL again
+            var textureCache = this.cache.getRessource(url);
+
+            if (textureCache !== undefined)
+                return Promise.resolve(textureCache);
+
+
+            // bug #74
+            //var limits = layer.tileMatrixSetLimits[coWMTS.zoom];
+            // if (!limits || !coWMTS.isInside(limits)) {
+            //     var texture = -1;
+            //     this.cache.addRessource(url, texture);
+            //     return Promise.resolve(texture);
+            // }
+            // -> bug #74
+
+            return this._IoDriver.read(url).then(function(result) {
+                if (result !== undefined) {
+
+                    //TODO USE CACHE HERE ???
+
+                    result.texture = this.getTextureFloat(result.floatArray);
+                    result.texture.generateMipmaps = false;
+                    result.texture.magFilter = THREE.LinearFilter;
+                    result.texture.minFilter = THREE.LinearFilter;
+
+                    // In RGBA elevation texture LinearFilter give some errors with nodata value.
+                    // need to rewrite sample function in shader
+                    //result.texture.magFilter = THREE.NearestFilter;
+                    //result.texture.minFilter = THREE.NearestFilter;
 
 
+                    this.cache.addRessource(url, result);
+
+                    return result;
+                } else {
+                    var texture = -1;
+                    this.cache.addRessource(url, texture);
+                    return texture;
+                }
+            }.bind(this));
         };
 
 
+        WMS_Provider.prototype.executeCommand = function(command){
+            //var service;
+            var destination = command.paramsFunction.destination;
+            var tile = command.requester;
+
+            if(destination === 1) {
```

Comment:
IMHO the test should depend solely on `layer.mimetype`:
* if it's a color image format (eg: png, jpg) use `getColorTexture`
* if it's an `xbil` format then use `getElevationTexture` (which should probably have a different name).